### ISSUE TITLE
SqueezeNet custom training - val_acc doesn't move up

### ISSUE THREAD

**Issue by mludvig** [OlafenwaMoses/ImageAI#260]
Hi, I'm running a custom prediction on a set of ca 500 images across all 4 supported network types. While for DenseNet, ResNet and InceptionV3 I get over 90% validation accuracy after ~ 20 epochs, for SqueezeNet it stays at 17% constant all the time. The only thing I changed between the runs was `.setModelTypeAsSqueezeNet()`, everything else was the same as for the other model types.

Do I need anything special to get reasonable training results from SqueezeNet?

I would expect at least a slight movements in the accuracy across 25 rounds...

```
Epoch 1/25
62/63 [============================>.] - ETA: 0s - loss: 1.9469 - acc: 0.1270  
Epoch 00001: val_acc improved from -inf to 0.16532, saving model to ./models/model_ex-001_acc-0.165323.h5
63/63 [==============================] - 25s 396ms/step - loss: 1.9469 - acc: 0.1270 - val_loss: 1.9458 - val_acc: 0.1653
Epoch 2/25
62/63 [============================>.] - ETA: 0s - loss: 1.9459 - acc: 0.1089 
Epoch 00002: val_acc improved from 0.16532 to 0.17742, saving model to ./models/model_ex-002_acc-0.177419.h5
63/63 [==============================] - 20s 323ms/step - loss: 1.9459 - acc: 0.1091 - val_loss: 1.9459 - val_acc: 0.1774
Epoch 3/25
62/63 [============================>.] - ETA: 0s - loss: 1.9459 - acc: 0.1452 
Epoch 00003: val_acc did not improve from 0.17742
63/63 [==============================] - 21s 336ms/step - loss: 1.9459 - acc: 0.1468 - val_loss: 1.9459 - val_acc: 0.1774
Epoch 4/25
62/63 [============================>.] - ETA: 0s - loss: 1.9459 - acc: 0.1552 
Epoch 00004: val_acc did not improve from 0.17742
63/63 [==============================] - 20s 314ms/step - loss: 1.9459 - acc: 0.1528 - val_loss: 1.9459 - val_acc: 0.1774
Epoch 5/25
62/63 [============================>.] - ETA: 0s - loss: 1.9459 - acc: 0.1532     
Epoch 00005: val_acc did not improve from 0.17742
63/63 [==============================] - 20s 315ms/step - loss: 1.9459 - acc: 0.1567 - val_loss: 1.9459 - val_acc: 0.1774
[...]
Epoch 23/25
62/63 [============================>.] - ETA: 0s - loss: 1.9459 - acc: 0.1371 
Epoch 00023: val_acc did not improve from 0.17742
63/63 [==============================] - 22s 341ms/step - loss: 1.9459 - acc: 0.1349 - val_loss: 1.9459 - val_acc: 0.1774
Epoch 24/25
62/63 [============================>.] - ETA: 0s - loss: 1.9459 - acc: 0.1492 
Epoch 00024: val_acc did not improve from 0.17742
63/63 [==============================] - 20s 319ms/step - loss: 1.9459 - acc: 0.1488 - val_loss: 1.9459 - val_acc: 0.1774
Epoch 25/25
62/63 [============================>.] - ETA: 0s - loss: 1.9459 - acc: 0.1310 
Epoch 00025: val_acc did not improve from 0.17742
63/63 [==============================] - 20s 316ms/step - loss: 1.9459 - acc: 0.1310 - val_loss: 1.9459 - val_acc: 0.1774

real	8m37.831s
user	29m23.175s
sys	0m28.460s
```

**Comment by OlafenwaMoses** [comment]
We will review this and give you an update on a fix.

**Comment by mludvig** [comment]
If it helps here is my training set and the training script: [training.zip](https://github.com/OlafenwaMoses/ImageAI/files/3469496/training.zip)

I only have 72 training images from each category, maybe that's not enough for SqueezeNet to learn from? All other algorithms work well though and give 98% accuracy. I'd love to use SqueezeNet as it seems to do the fastest inference on Raspberry Pi.

**Comment by Multhree** [comment]
I met the same problem, and I change the batch_size from 32 to 128, then the val_acc move up.

**Comment by manueltarouca** [comment]
I have the same problem, with the idenprof dataset

**Comment by cchimbooo** [comment]
I'm having the exact same problem here :s
